app/filtering.js

```diff
@@ -273,9 +271,15 @@ function registerForBeforeSendHeaders (session, partition) {
 
       if (cookieSetting === 'blockAllCookies' ||
         isThirdPartyHost(parsedFirstPartyUrl.hostname, parsedTargetUrl.hostname)) {
+        let hasCookieException = false
+        cookieExceptions.forEach((exceptionPair) => {
```

Comment:
i think this would be more efficient if cookieExceptions were a map like `{firstPartyOrigin: [origins]}` instead of a list like `[[firstPartyOrigin, origin],...]`